File: src/problems/mario/GetHeight.java
import java.util.Scanner;

public class GetHeight {
    
    public static int height() {
        Scanner input = new Scanner(System.in);
        int height = 0;

        do {
            System.out.print("Insert a number: ");
            if (input.hasNextInt()) {
                height = input.nextInt();
            } else {
                input.nextLine(); 
                height = 0;
            }
        } while (height < 1 || height > 8);

        input.close();
        return height;
    }
}

File: src/problems/mario/get_height.c
#include <stdio.h>
#include "get_height.h"

int get_height() {
    int height = 0;
    do {
        printf("Insert a number: ");
        if (scanf("%d", &height) != 1) {
            while (getchar() != '\n');
            height = 0;
        }
    } while (height < 1 || height > 8);
    return height;
}

